socsec/otptool.py

```diff
@@ -1141,9 +1141,15 @@ def make_scu_protect_image(self, scu_protect_config, strap_info):
             bytearray(scu_ignore.tobytes())
 
     def make_otp_image(self, config_file, key_folder,
-                       user_data_folder, output_folder, no_last_bit=False):
+                       user_data_folder, output_folder,
+                       no_last_bit=False, no_pre_production=False):
         otp_config = jstyleson.load(config_file)
 
+        if no_pre_production:
+            if otp_config['version'] != 'A3' and \
+              otp_config['version'] != '1030A2':
```

Comment:
I think you should probably define this condition in terms of the set of pre-production SoCs. That way if you produce an A4 (maybe? hopefully not?) this doesn't incorrectly fail. So:

```
if opt_config['version'] in ['A0', 'A1', 'A2', '1030A0', '1030A1']:
 raise OtpError(...)
```